src/crimson/mgr/client.cc

```diff
@@ -96,12 +95,11 @@ void Client::ms_handle_reset(crimson::net::ConnectionRef c, bool /* is_replace *
 {
   LOG_PREFIX(Client::ms_handle_reset);
   DEBUGDPP("", *this);
-  gates.dispatch_in_background(__func__, *this, [this, c] {
+  gates.dispatch_in_background(__func__, *this,
+  [this, c]() -> seastar::future<> {
     if (conn == c) {
       report_timer.cancel();
-      return reconnect();
-    } else {
-      return seastar::now();
+      co_await reconnect();
```

Comment:
IIRC, if `c` is used after this co_await, we might hit undefined behaviour, because `c` may have already been destroyed. So I still think we should avoid use coroutines in lambdas that captures something.

Reply:
Missed that, thanks!